task/power/src/main.rs

```diff
@@ -692,7 +692,7 @@ impl ServerImpl {
             .ok_or(ResponseCode::NoDevice)?;
         let (dev, rail) = (cfg.builder)(self.i2c_task);
 
-        #[repr(packed)]
+        #[repr(C, packed)]
```

Comment:
Where did the `repr(C)` come from? (I feel like you mentioned this in chat but I can't remember)

Reply:
I didn't, this was a relatively new discovery --- there's now [a lint for `repr(packed)` without specifying an ABI](https://rust-lang.github.io/rust-clippy/stable/index.html#repr_packed_without_abi), because it defaults to `repr(Rust, packed)`, which lacketh a stable ABI. I thought that since we are essentially hand-implementing `zerocopy` here, we should ensure the layout is stable across compilation units, although it may not actually matter here.